README.md

```diff
@@ -2086,6 +2086,35 @@ params do
 end
 ```
 
+## Parameter Validation and Coercion using `dry-schema`
+
+As an alternative to the `params` DSL described above, you can use a
+schema or `dry-validation` contract to describe an endpoint's
+parameters. Call `contract` with a schema defined previously
+
+```rb
+contract CreateFoosSchema
+```
+
+or with a block, using the [schema definition
+syntax](https://dry-rb.org/gems/dry-schema/1.13/#quick-start):
+
+```rb
+contract do
+  required(:foos).array do
+    required(:name).filled(:string)
+    optional(:volume).maybe(:integer, lt?: 9)
+  end
+end
+```
+
+The latter will define a coercing schema (`Dry::Schema.Params`). With
+the former syntax it's up to you to decide whether the input will need
+coercing.
+
+`params` and `contract` declarations can also be used together, e.g.
+to describe different parts of a nested namespace for an endpoint.
+
```

Comment:
You have to say to add `dry-schema` to Gemfile here, right? We should probably have an integration spec without it as well to make sure we don't accidentally rely on something from that gem that only surfaces in production.

Reply:
Indeed, if we don't declare `dry-schema` a hard dependency of `grape`.

> We should probably have an integration spec without it as well to make sure we don't accidentally rely on something from that gem that only surfaces in production.

How do you suggest it be written? Probably not with a separate Gemfile. Unbinding the constant `Dry` temporarily?

Reply:
I see that now: the Appraisal config, with indeed separate Gemfiles for certain integration test files.

The instructions in the root Gemfile do not quite work, however:

```sh
appraisal install ; rubocop -a gemfiles/*.gemfile`
```

both fails at the `rails_5` gemfile (a screenful about activesupport versions), generates some `.gemfile` files that weren't there before (apparently desynchronization happened between `rack_1_0.gemfile` and `rack1.gemfile` naming), and creates gemfiles with lots of double quotes inside them. Which in turn tiggers Rubocop for each of those lines.

What do you recommend I do here?

Regarding the integration testing itself, I think we'll be more or less safe as long as at least some of those gemfiles (the ones used for existing integration tests, for example) don't include `dry-schema`.

Reply:
Okay, I fixed the above, and indeed those existing test won't help. Looking into adding a new one.